### DISCUSSION TITLE
Interested in making a contribution

### DISCUSSION THREAD

**Discussion by AndreasH96** [gyscos/cursive#688] [Ideas]
Hi! 

I've started my journey towards learning Rust and decided to create tetris with your nice library :smile: 
I'm in the need of listening to if the player presses the Space Bar and from what I can see, it isn't implemented in the Events. 
I'm interested in adding it myself if that's OK with you? 
I'm not sure how big of a change it would be, but I expect that it's quite small. 
Could you point me towards the workflow of adding it? :smile:

**Comment by gyscos** [comment]
Hi, and thanks for the question!
Spacebar should sent a `Event::Char(' ')` character.
https://docs.rs/cursive/latest/cursive/event/enum.Event.html#variant.Char

You can run the `key_codes` example, it just prints the events it receives. This way you can see what is supported and how to process it (arrows, Fn keys, alt/shift/ctrl modifiers, ...).

Now, to answer the question (even though the change is not actually required):
* First, the "event model" need to support the event. Here it's `Event::Char(' ')`, but for some new event you may need to add a new entry to the [`Event`](https://github.com/gyscos/cursive/blob/main/cursive-core/src/event.rs#L486) enum.
* Next, you would need to update the various backends to actually detect this and send the proper events.
  * For example, [this is currently](https://github.com/gyscos/cursive/blob/main/cursive/src/backends/crossterm.rs#L112-L115) where the crossterm backend parsed "crossterm char events" and translate them to "cursive char events". For a new event, you would need to find out if the backend library (here crossterm) supports it, and how they send this info.
  * Each backend has its own way to represent and communicate events to cursive, so this work would need to be repeated for all other backends that support it (currently ncurses, pancurses, termion, bear-lib-terminal).

**Comment by AndreasH96** [comment]
Yupp, I should have understood that it can be done with `Event::Char(' ')`. Thank you for the quick answer! :smile: 
Thank you for explanation!
